### ISSUE TITLE
UPDATE https://github.com/PHPMailer/PHPMailer/wiki/Using-Gmail-with-XOAUTH2

### ISSUE THREAD

**Issue by batata004** [PHPMailer/PHPMailer#2691]
Now that Google does not allow LESS SECURED APPS I am having a pretty hard time following the manual at:

https://github.com/PHPMailer/PHPMailer/wiki/Using-Gmail-with-XOAUTH2

All the screens are completely different now, the options are different... despite that, with a lot of effort (I am not a good programmer) I managed to follow most of the steps of the link above (creating project, app, consent...). HOWEVER everytime I execute the final php script it shows this error below:

```
Fatal error: Uncaught GuzzleHttp\Exception\RequestException: cURL error 60: SSL certificate problem: unable to get local issuer certificate (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for https://oauth2.googleapis.com/token in Z:\htdocs\teste\vendor\guzzlehttp\guzzle\src\Handler\CurlFactory.php:211 Stack trace:
 #0 Z:\htdocs\teste\vendor\guzzlehttp\guzzle\src\Handler\CurlFactory.php(158): GuzzleHttp\Handler\CurlFactory::createRejection(Object(GuzzleHttp\Handler\EasyHandle), Array)
#1 Z:\htdocs\teste\vendor\guzzlehttp\guzzle\src\Handler\CurlFactory.php(110): GuzzleHttp\Handler\CurlFactory::finishError(Object(GuzzleHttp\Handler\CurlHandler), Object(GuzzleHttp\Handler\EasyHandle), Object(GuzzleHttp\Handler\CurlFactory)) 
#2 Z:\htdocs\teste\vendor\guzzlehttp\guzzle\src\Handler\CurlHandler.php(47): GuzzleHttp\Handler\CurlFactory::finish(Object(GuzzleHttp\Handler\CurlHandler), Object(GuzzleHttp\Handler\EasyHandle), Object(GuzzleHttp\Handler\CurlFactory)) 
#3 Z:\htdocs\teste\vendor\guzzlehttp\guzzle\src\Handler\Proxy.php(28): GuzzleHttp\Handler\CurlHandler->__invoke(Object(GuzzleHttp\Psr7\Request), Array) 
#4 Z:\htdocs\teste\vendor\guzzlehttp\guzzle\src\Handler\Proxy.php(48): GuzzleHttp\Handler\Proxy::GuzzleHttp\Handler\{closure}(Object(GuzzleHttp\Psr7\Request), Array) 
#5 Z:\htdocs\teste\vendor\guzzlehttp\guzzle\src\PrepareBodyMiddleware.php(64): GuzzleHttp\Handler\Proxy::GuzzleHttp\Handler\{closure}(Object(GuzzleHttp\Psr7\Request), Array) 
#6 Z:\htdocs\teste\vendor\guzzlehttp\guzzle\src\Middleware.php(31): GuzzleHttp\PrepareBodyMiddleware->__invoke(Object(GuzzleHttp\Psr7\Request), Array) 
#7 Z:\htdocs\teste\vendor\guzzlehttp\guzzle\src\RedirectMiddleware.php(71): GuzzleHttp\Middleware::GuzzleHttp\{closure}(Object(GuzzleHttp\Psr7\Request), Array) #8 Z:\htdocs\teste\vendor\guzzlehttp\guzzle\src\Middleware.php(63): GuzzleHttp\RedirectMiddleware->__invoke(Object(GuzzleHttp\Psr7\Request), Array) 
#9 Z:\htdocs\teste\vendor\guzzlehttp\guzzle\src\HandlerStack.php(75): GuzzleHttp\Middleware::GuzzleHttp\{closure}(Object(GuzzleHttp\Psr7\Request), Array) 
#10 Z:\htdocs\teste\vendor\guzzlehttp\guzzle\src\Client.php(331): GuzzleHttp\HandlerStack>__invoke(Object(GuzzleHttp\Psr7\Request), Array) 
#11 Z:\htdocs\teste\vendor\guzzlehttp\guzzle\src\Client.php(107): GuzzleHttp\Client->transfer(Object(GuzzleHttp\Psr7\Request), Array) 
#12 Z:\htdocs\teste\vendor\guzzlehttp\guzzle\src\Client.php(123): GuzzleHttp\Client->sendAsync(Object(GuzzleHttp\Psr7\Request), Array) 
#13 Z:\htdocs\teste\vendor\league\oauth2-client\src\Provider\AbstractProvider.php(608): GuzzleHttp\Client->send(Object(GuzzleHttp\Psr7\Request)) 
#14 Z:\htdocs\teste\vendor\league\oauth2-client\src\Provider\AbstractProvider.php(621): League\OAuth2\Client\Provider\AbstractProvider->getResponse(Object(GuzzleHttp\Psr7\Request)) 
#15 Z:\htdocs\teste\vendor\league\oauth2-client\src\Provider\AbstractProvider.php(537): League\OAuth2\Client\Provider\AbstractProvider->getParsedResponse(Object(GuzzleHttp\Psr7\Request))
#16 Z:\htdocs\teste\teste.php(140): League\OAuth2\Client\Provider\AbstractProvider->getAccessToken(Object(League\OAuth2\Client\Grant\AuthorizationCode), Array) 
#17 {main} thrown in Z:\htdocs\teste\vendor\guzzlehttp\guzzle\src\Handler\CurlFactory.php on line 211
```

Before this error, I managed to fix other 2 errors taking me almost 4 hours, installing dependencies on Composer that were not mentioned in the phpmailer mailer documentation. But now I am stuck at this error above.

Many many many more people will face this problem from now on because of the very recent change on Google regarding less secure apps. So I think someone with great knowledge in programming could fix the manual above, because it's full of problems...

The error above I am almost sure is because my localhost does not run on https... so how do I fix that? I discovered that I can set verifyssl to false but I have no idea where I should put that code.

**Comment by Synchro** [comment]
It's always worth searching for the error message. In this case the top hit was [this SO question](https://stackoverflow.com/questions/28858351/php-ssl-certificate-error-unable-to-get-local-issuer-certificate), which covers the same ground as the PHPMailer troubleshooting guide on this subject. Both say you need to download an updated CA certificate bundle, or update your server's packages. Note that has nothing to do with your own server's TLS certificate, and no, you don't need your callback URL to be on HTTPS – you are the only one who will ever access it, so HTTP to localhost will work just fine.

It would help if you said which dependencies you're talking about.

I wrote that guide a long time ago, and I know it's outdated, but this is open source, so if you see something wrong, figure out how you can help to fix it. It's mostly screenshots and label descriptions that will need updating, because the fundamental flows of OAuth 2 and Google's authentication have not changed at all.

You may find it easier to create an _app password_ in gmail (search for that term). Then you can just go back to using regular password auth, and it still works without needing the obsolete "less secure apps" option.

**Comment by batata004** [comment]
Hi my friend, first of all sorry if I looked rude. I really appreciate all your kindness providing such a tutorial. However I had many problemas following it, many errors executing the php script in the tutorial because of lack of libraries (I managed to fix those but I dont even remember which ones I had to download)... and I am not it covering the problems regarding the interface cause many things are now very different to setup inside google cloud.

I thank you for explaining the problem related to the certificate, however that looks too complicated to me and I will probably have to deal with a few more problem on the way to finally be able to allow the account to use oauth. Google made it so complicated.

The suggestion you provided on your last paragraph I think is the best alternative. I recommend you to add that suggestion to the tutorial!

Despite all of this, I decided to use another mail provider that allows login with SMTP (Zoho) and I managed to fix it in no more than 8 minutes and everything is working fine. Thank you for your help my friend.
